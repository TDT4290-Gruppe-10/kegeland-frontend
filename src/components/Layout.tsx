import { Box, Flex, useMediaQuery } from '@chakra-ui/react';
import { useEffect, useState } from 'react';

import withAuthUser, { WithAuthUserProps } from '../hoc/withAuthUser';

import Header from './Header';
import Sidebar from './Sidebar';

type LayoutProps = {
  children?: React.ReactNode;
} & WithAuthUserProps;

const Layout: React.FC<LayoutProps> = ({ user, children }) => {
  const [isGreaterThanMd] = useMediaQuery('(min-width: 48em)');
  const [openSidebar, setOpenSidebar] = useState<boolean>(true);

  useEffect(() => {
    if (!isGreaterThanMd && openSidebar) {
      setOpenSidebar(false);
    }

    if (isGreaterThanMd && !openSidebar) {
      setOpenSidebar(true);
    }
  }, [isGreaterThanMd]);

  const toggle = () => {
    setOpenSidebar(!openSidebar);
  };

  return (
    <Box h="100%" w="100%">
      <Flex flexDir="row">
        <Box position={isGreaterThanMd ? 'relative' : 'absolute'}>
          <Sidebar user={user} isOpen={openSidebar} />
        </Box>

        <Flex flexDir="column" flex={1} flexGrow={1}>
          <Header toggleSidebar={toggle} />
          <Box
            zIndex={999}
            flexDir="column"
            padding={5}
            marginX="auto"
            overflow="hidden"
            w="100%"
            maxW={{
              base: 'container.sm',
              md: 'container.md',
              xl: 'container.lg',
            }}>
            {children}
          </Box>
        </Flex>
      </Flex>
    </Box>
  );
};

export default withAuthUser(Layout);
