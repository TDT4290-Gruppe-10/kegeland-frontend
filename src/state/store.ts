import { combineReducers, configureStore } from '@reduxjs/toolkit';
import logger from 'redux-logger';
import {
  persistStore,
  persistReducer,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from 'redux-persist';
import storage from 'redux-persist/lib/storage';

import authReducer from './ducks/auth/auth.reducer';
import {patientReducer, singlePatientReducer} from './ducks/patients/patients.reducer';
import { sensorReducer } from './ducks/sensors/sensor.reducer';

const persistConfig = {
  key: 'root',
  version: 1,
  storage: storage,
  whitelist: ['auth'],
};

const rootReducer = combineReducers({
  auth: authReducer,
  patients: patientReducer,
  singlePatient: singlePatientReducer,
  sensorData: sensorReducer
});

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
  reducer: persistedReducer,
  middleware: (getDefaultMiddleware) => {
    return getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
      },
    })
      .concat(logger)
  },
  devTools: process.env.NODE_ENV !== 'production',
});

export const persistor = persistStore(store);

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
