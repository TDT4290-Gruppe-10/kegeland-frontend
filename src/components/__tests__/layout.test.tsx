import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import Layout from '../Layout';
import { store } from '../../state/store';

describe('Test layout', () => {
  it('renders correctlyu', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <Layout />
          </MemoryRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
