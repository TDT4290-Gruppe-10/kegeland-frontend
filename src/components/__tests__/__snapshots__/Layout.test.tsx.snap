// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test layout renders correctlyu 1`] = `null`;
